### PR TITLE
[pg] Add support for sslmode parameter

### PR DESCRIPTION
**Pull request by ikokostya**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/brianc/node-postgres/blob/60d8df659c5481723abada2344ac14d77377338c/lib/connection-parameters.js#L108
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by ikokostya** [comment]
@pspeter3 Could you please review pr?

File: types/pg/index.d.ts [hunk 1]
```diff
@@ -33,8 +34,16 @@ export interface Defaults extends ConnectionConfig {
 
 import { ConnectionOptions } from "tls";
 
+// Note: modes 'disable' and 'allow' don't work correctly. If property `ssl` is not undefined, SSL
```
Comment (pspeter3) [inline types/pg/index.d.ts]: Are they types supported by the library though? I worry about editorializing the library with the type definition.

**Review by pspeter3 — APPROVED** [review]

**Comment by ikokostya** [comment]
I closing PR for now, because of need more investigation. The function [`getLibpqConnectionString`](https://github.com/brianc/node-postgres/blob/60d8df659c5481723abada2344ac14d77377338c/lib/connection-parameters.js#L99) is used only by [native client](https://github.com/brianc/node-postgres/search?utf8=%E2%9C%93&q=getLibpqConnectionString&type=).

**Comment by ikokostya** [comment]
> I closing PR for now, because of need more investigation. The function getLibpqConnectionString is used only by native client.

So, yes, this PR was incorrect. In JavaScript library `ssl` can be only `boolean` or `TLS` options.

**Pull request closed without merging**